spec/controllers/scim_rails/scim_users_controller_spec.rb

```diff
@@ -364,9 +365,35 @@
 
         expect(response.status).to eq 201
         expect(company.users.count).to eq 1
-        user = company.users.first
         expect(user.archived?).to eq true
       end
+
+      it "creates user while with emails corresponding to their types" do
+        post :create, params: {
+          id: 1,
+          userName: "test@example.com",
+          name: {
+            givenName: "Test",
+            familyName: "User",
+          },
+          emails: [
+            {
+              type: "other",
+              value: "other@example.com",
+            },
+            {
+              type: "work",
+              value: "work@example.com",
+            },
+          ]
+        }
+
+        expect(response.status).to eq(201)
+        expect(company.users.count).to eq(1)
+
+        expect(user.email).to eq("work@example.com")
+        expect(user.alternate_email).to eq("other@example.com")
```

Comment:
🌵 I'd make these be `var`s or `let`s so you don't have to retype an expected string

Reply:
These were actually the tests that were already here when we forked this repo, I will make another PR that refactors this whole file altogether